src/mp2_ri_gpw.F

```diff
@@ -1469,32 +1534,65 @@ SUBROUTINE mp2_update_P_gamma(mp2_env, para_env_sub, gd_B_virtual, &
 
       CALL dgemm_counter_start(dgemm_counter)
       IF (.NOT. (alpha_beta)) THEN
+#if defined(__SPLA)
+         spla_error_code = spla_dgemm(SPLA_OP_TRANSPOSE, SPLA_OP_NONE, my_B_size(ispin), my_B_size(ispin), virtual(ispin), 1.0_dp, &
+                                      c_loc(t_ab), virtual(ispin), c_loc(local_ab), virtual(ispin), &
+                                      1.0_dp, c_loc(mp2_env%ri_grad%P_ab(ispin)%array(:, &
+                                                                              my_B_virtual_start(ispin):my_B_virtual_end(ispin))), &
+                                      my_B_size(ispin), mp2_env%spla_ctx)
+#else
```

Comment:
Is it possible to shift this dependence on SPLA to an external module which makes it more reusable to other parts of the code?

Reply:
And would be the point of adding yet another interface to an interface. The only thing needed would be to move the spla context one step up if you want to use it elsewhere. Right now the context is located in the mp2 structure. 

Moreover I do not want to overload the dgemm function because offloading is not always needed.

Reply:
If you are allowed to have only one SPLA structure, I prefer to have it at a higher level then such that a developer who wants to use SPLA elsewhere does not run into conflicts with the MP2 code.